app/User.php

```diff
@@ -82,4 +90,29 @@ public function getFollowings ()
         }
         return $followings;
     }
+
+    public function getActivities()
+    {
+        $followings = $this->getFollowings();
+        $activities = $this->activities;
+        foreach ($followings as $following) {
+            $activities = $activities->merge($following->activities);
+        }
+        $activities = $activities->sortByDesc('id');     
+        $subjects = [];
+        $actions = [];
+        $objects = [];
+        $times = [];
+        foreach ($activities as $activity) {
+            array_push($subjects, User::find($activity->user_id));
```

Comment:
- you dont need to call User inside a User model. you can call it by `$this`.
- instead of using array try to use collections in laravel.

Reply:
About $object, it includes User model data and Category model data. Because an object of an action could be both like "Tom followed Jem (User model data)" and "Tom learned 5 of 5 words in Basic (Category model data)".

Is it possible to replace array to collection about $object?